### DISCUSSION TITLE
Will useQuery variable state change makes the query refetched?

### DISCUSSION THREAD

**Discussion by Trav1sRen** [urql-graphql/urql#2209] [Q&A] [answered]
I am using GraphQL Code Generator with urql.
Now I have following code,
`
const [variables, setVariables] = useState<PostsQueryVariables>({ limit: 10 });
`
`
const [{ data, fetching }] = usePostsQuery({
    variables
});
`

The code is using the Component State as the query variable.
My question is, when the state changes, will the query be refetched.

I understood this hook as a combination of `useEffect` and `client.query()`,
but I am not sure if the `variable` is put into the `useEffect` dependencies list.

 I tried to check the source code, but it is a bit complex to understand, as there are some computing upon the `deps`.
Could someone help me to understand this?

**Comment by JoviDeCroock** [comment, marked as answer]
It's put in there as we recalculate the request with useRequest the key generated there is a unique identifier for query + variables and is the primary way to initiate a new stream

**Comment by Trav1sRen** [comment]
Thanks!
